server/routes/events.js

```diff
@@ -12,101 +13,507 @@ router.get('/ping', function(req, res, next) {
+																		)
+																		.del();
+																}
+															})
+															.then(trx.commit)
+													})
+											})
+									})
+							})
+					})
+  		})
+  		.catch(trx.rollback);
+  })
+  .then(function() {
+  	res.send(util.message('Successfully updated event_id: ' + event_id));
+  })
+  .catch(err => { return next(err) });
+});
+
+
+// Delete an event by event_id
+router.delete('/:event_id', function(req, res, next) {
+  const event_id = req.params.event_id;
+  knex('event').del().where({ fb_id: event_id, }).then(result => {
+    if (result) {
+      res.send(util.message('Successfully deleted event with event_id ' + event_id));
+    } else {
+      util.throwError(404, 'No event found to delete');
+    }
+  })
+  .catch(err => { return next(err) });
 });
 
 // GET all users that have favorited the given event
 router.get('/:event_id/favorites', function(req, res, next) {
   const event_id = req.params.event_id;
-  db.query('SELECT user_id FROM favorite_events WHERE event_id = $1', [event_id], (err, result) => {
-		if (err) return next(err);
-  	res.send(result.rows);
-  });
+  knex('favorite_events').select('user_id')
+  .where({ event_id: event_id })
+  .then(result => {
+    if(result.length) {
+      res.json(result);
+    } else {
+      res.status(404).json('No events where fb_id = ' + event_id + ' found.');
+    }
+  })
 });
 
-// GET all users registered and have paid or not paid for a given event
+// Update user registration for a given event
+router.put('/:event_id/register/:user_id', function(req, res, next) {
+  const event_id = req.params.event_id;
+  const user_id = req.params.user_id;
+  const paid = req.query.paid;
+  knex('event_registration')
+  .update({ 
+    has_paid: paid, 
+    updated_at: knex.raw('now()')
+  })
+  .where({ 
+    event_id: event_id, 
+    user_id: user_id
+  })
+  .then(result => {
+    if (result) {
+      res.send(util.message('Successfully updated user registration for event with event_id ' + event_id));
+    } else {
+      util.throwError(404, 'No user found with user_id ' + user_id + ' for event with event_id ' + event_id);
+    }
+  })
+  .catch(err => { return next(err) });
+});
+
+// // GET all users registered and have paid or not paid for a given event
 router.get('/:event_id/register', function(req, res, next) {
   const event_id = req.params.event_id;
   const paid = req.query.paid;
   if (paid == undefined) {
-		db.query('SELECT user_id FROM event_registration WHERE event_id = $1', [event_id], (err, result) => {
-	  	if (err) return next(err);
-    	res.send(result.rows);
-  	});
+    knex('event_registration').select('user_id')
+    .where({ event_id: event_id })
+    .then(result => {
+    if(result.length) {
+      res.json(result);
+    } else {
+      res.status(404).json('No events where fb_id = ' + event_id + ' found.');
+    }
+  })
   } else if (paid == "true" || paid == "false") {
```

Comment:
Can simplify/deduplicate code here. If it's true, false, or null I believe you can just do `has_paid: paid` and knex will handle it accordingly. If not one of those values, don't make a query